helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -369,11 +389,11 @@ private void carryOverPreviousCurrentState() {
         }
 
         StateModelDefinition stateModel =
-            _dataAccessor.getProperty(_keyBuilder.stateModelDef(stateModelDefRef));
+            dataAccessor.getProperty(keyBuilder.stateModelDef(stateModelDefRef));
 
-        BaseDataAccessor<ZNRecord> baseAccessor = _dataAccessor.getBaseDataAccessor();
+        BaseDataAccessor<ZNRecord> baseAccessor = dataAccessor.getBaseDataAccessor();
         String curStatePath =
-            _keyBuilder.currentState(_instanceName, _sessionId, lastCurState.getResourceName())
+            keyBuilder.currentState(instanceName, sessionId, lastCurState.getResourceName())
                 .getPath();
 
         String initState = stateModel.getInitialState();
```

Comment:
This require some options here:
1. If reset happens, it carries the initial state.
2. If last exist of freeze mode, it requires carrying over with original state without resetting.